frontend/src/metabase/lib/entity-id/hooks/use-validated-entity-id.ts

```diff
@@ -64,11 +64,19 @@ export const useValidatedEntityId = <
     }
 
     if (isLoading) {
-      return { id: null, isLoading: true, isError: false } as const;
+      return {
+        id: null,
+        isLoading: true,
+        isError: false,
+      } as const;
     }
 
-    if (isError) {
-      return { id: null, isLoading: false, isError: true } as const;
+    if (!isEntityId || isError) {
```

Comment:
if it's not an entity ID here, there won't be `isError` since we will not call the endpoint to resolve an entity ID in the first place, so we need this extra condition.